tests/templates/test_subroutines/test_qubitization.py

```diff
@@ -0,0 +1,308 @@
+
+
+@pytest.mark.parametrize(
+    "hamiltonian, expected_unitaries",
+    (
+        (
+            qml.ops.LinearCombination(
+                np.array([1, -1, 2]), [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+            ),
+            [
+                qml.PauliX(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+                qml.PauliY(0) @ qml.GlobalPhase(np.array(np.pi), wires=0),
+                qml.PauliZ(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+            ],
+        ),
+        (
+            qml.ops.LinearCombination(
+                np.array([1.0, 1.0, 2.0]), [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+            ),
+            [
+                qml.PauliX(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+                qml.PauliY(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+                qml.PauliZ(0) @ qml.GlobalPhase(np.array([0.0]), wires=0),
+            ],
+        ),
+        (
+            qml.ops.LinearCombination(
+                np.array([-0.2, -0.6, 2.1]), [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0)]
+            ),
+            [
+                qml.PauliX(0) @ qml.GlobalPhase(np.array(np.pi), wires=0),
+                qml.PauliY(0) @ qml.GlobalPhase(np.array(np.pi), wires=0),
+                qml.PauliZ(0) @ qml.GlobalPhase(np.array(0), wires=0),
+            ],
+        ),
+    ),
+)
+def test_positive_coeffs_hamiltonian(hamiltonian, expected_unitaries):
+    """Tests that the function _positive_coeffs_hamiltonian correctly transforms the Hamiltonian"""
+
+    new_coeffs, new_unitaries = _positive_coeffs_hamiltonian(hamiltonian)
+
+    assert np.allclose(new_coeffs, np.abs(hamiltonian.terms()[0]))
+
+    for i, unitary in enumerate(new_unitaries):
+        assert qml.equal(expected_unitaries[i], unitary)
+
+
+@pytest.mark.parametrize(
+    "hamiltonian",
+    [
+        qml.dot([0.2, -0.5, 0.3], [qml.Y(0) @ qml.X(1), qml.Z(1), qml.X(0) @ qml.Z(2)]),
+        qml.dot([0.3, -0.5, 0.3], [qml.Z(0) @ qml.X(1), qml.X(1), qml.X(0) @ qml.Y(2)]),
+        qml.dot([0.4, -0.5, -0.3], [qml.Z(0) @ qml.X(2), qml.Y(0), qml.X(1) @ qml.Z(2)]),
+    ],
+)
+def test_operator_definition_qpe(hamiltonian):
+    """Tests Check that we can obtain the eigenvalues of H by applying the Qubitization Operator."""
+
+    from scipy.signal import find_peaks
+
+    @qml.qnode(qml.device("default.qubit"))
+    def circuit(theta):
+
+        # initial state
+        qml.RX(theta[2], wires=0)
+        qml.CRY(theta[3], wires=[0, 2])
+        qml.RY(theta[0], wires=2)
+        qml.CRY(theta[4], wires=[1, 2])
+        qml.RX(theta[1], wires=1)
+        qml.CRX(theta[2], wires=[2, 0])
+
+        # apply QPE (used iterative qpe here)
+        measurements = qml.iterative_qpe(
+            qml.Qubitization(hamiltonian, control=[3, 4]), ancilla=5, iters=8
+        )
+
+        return qml.probs(op=measurements)
+
+    theta = np.array([0.1, 0.2, 0.3, 0.4, 0.5]) * 100
```

Comment:
Why `* 100`?

Reply:
In order to make it more random, (>> 2pi ) It can be remove